File: app/src/main/java/com/example/mysmartschool/Guru/api/ApiClient.java
package com.example.mysmartschool.Guru.api;

import com.example.mysmartschool.Config;

import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.logging.HttpLoggingInterceptor;

public class ApiClient {
    public OkHttpClient resourceClient() {
        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
        interceptor.setLevel(HttpLoggingInterceptor.Level.HEADERS);
        interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
        OkHttpClient.Builder client = new OkHttpClient.Builder();
        client.addInterceptor(chain -> {
            Request req = chain.request();
            Request newReq;
            newReq = req.newBuilder()
                    .header("Authorization", String.format("Bearer %s", Config.API_TOKEN))
                    .build();
            return chain.proceed(newReq);
        });
//        client.addInterceptor(interceptor);
        return client.build();
    }
}

File: app/src/main/java/com/example/mysmartschool/Guru/api/ApiClient.kt
package com.example.mysmartschool.Guru.api

import com.example.mysmartschool.Config
import okhttp3.*
import okhttp3.logging.HttpLoggingInterceptor
import java.util.concurrent.TimeUnit

fun loginClient(): OkHttpClient {
    val logInterceptor = HttpLoggingInterceptor()
    logInterceptor.level = HttpLoggingInterceptor.Level.BODY
    val builder = OkHttpClient.Builder()
    builder.readTimeout(Config.REQUEST_TIMEOUT.toLong(), TimeUnit.SECONDS)
    builder.connectTimeout(Config.REQUEST_TIMEOUT.toLong(), TimeUnit.SECONDS)
    builder.addInterceptor(logInterceptor)
    builder.addInterceptor(object : Interceptor {
        override fun intercept(chain: Interceptor.Chain): Response {
            val req = chain.request().newBuilder()
            req.addHeader("Accept", "application/json")
            return chain.proceed(req.build())
        }

    })

    return builder.build()
}

fun resourceClient(): OkHttpClient {
    val logInterceptor = HttpLoggingInterceptor()
    logInterceptor.level = HttpLoggingInterceptor.Level.BODY
    val builder = OkHttpClient.Builder()
    builder.readTimeout(Config.REQUEST_TIMEOUT.toLong(), TimeUnit.SECONDS)
    builder.connectTimeout(Config.REQUEST_TIMEOUT.toLong(), TimeUnit.SECONDS)
    builder.addInterceptor(logInterceptor)
    builder.addNetworkInterceptor(object : Interceptor {
        override fun intercept(chain: Interceptor.Chain): Response {
            val req = chain.request().newBuilder()
            req.addHeader("Accept", "application/json")
            req.addHeader("Authorization",String.format("Bearer %s", Config.API_TOKEN))
            return chain.proceed(req.build())
        }
    })
    builder.authenticator(object : Authenticator {
        override fun authenticate(route: Route?, response: Response): Request? {
            return response.request.newBuilder().header("Authorization",String.format("Bearer %s", Config.API_TOKEN)).build()
        }
    })
    return builder.build()
}